src/Analyzer.JsonRuleEngine.UnitTests/InOperatorTests.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.Operators;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.UnitTests
+{
+    [TestClass]
+    public class InOperatorTests
+    {
+        [DataTestMethod]
+        [DataRow(true, new object[] { "anotherValue", "aValue" }, DisplayName = "String is in string array")]
+        [DataRow(false, new object[] { "anotherValue", "otherValue" }, DisplayName = "String is not in string array")]
+        [DataRow(false, new object[] { }, DisplayName = "String is not in empty array")]
+        [DataRow(true, new object[] { "anotherValue", 4, false, 3.5, null, "aValue" }, DisplayName = "String is in mixed array")]
+        [DataRow(false, new object[] { "anotherValue", 4, false, 3.5, null, "otherValue" }, DisplayName = "String is not in mixed array")]
+        public void EvaluateExpression(bool evaluationResult, params object[] arrayOfValues)
+        {
+            var aValue = "aValue";
+
+            var valueJToken = JsonRuleEngineTestsUtilities.ToJToken(aValue);
+            var arrayOfValuesJToken = JsonRuleEngineTestsUtilities.ToJToken(arrayOfValues);
+
+            var inOperator = new InOperator(arrayOfValuesJToken, isNegative: false);
+            Assert.AreEqual(evaluationResult, inOperator.EvaluateExpression(valueJToken));
+        }
+
+        [TestMethod]
+        public void GetName_ReturnsCorrectName()
+        {
+            Assert.AreEqual("In", new InOperator(new JObject(), false).Name);
+        }
+    }
```

Comment:
Add tests for the constructor as well.

Reply:
I'm assuming you wrote this because of the array validation issue. I added a test for that in ExpressionConverterTests. It doesn't even compile if I try to use a non JArray as parameter in the constructor